scheduler/test/cook/test/kubernetes/controller.clj

```diff
@@ -103,7 +108,21 @@
     (is (nil? (do-process :missing :pod/failed)))
     (is (= :missing (do-process :missing :pod/running)))
     (is (nil? (do-process :missing :pod/unknown)))
-    (is (= :missing (do-process :missing :pod/waiting)))))
+    (is (= :missing (do-process :missing :pod/waiting)))
+    (is (nil? (do-process :cook-expected-state/running :missing :custom-test-state {:state :missing
+                                                                                    :reason "Node preempted"
+                                                                                    :pod-deleted? true
+                                                                                    :pod-preempted? true})))
+    (is (= :reason-slave-removed @reason))
+    (is (nil? (do-process :cook-expected-state/running :missing :custom-test-state {:state :missing
+                                                                                    :reason "Pod was explicitly deleted"
+                                                                                    :pod-deleted? true}
+                          :force-nil-pod? true)))
+    (is (= :reason-executor-terminated @reason))
+    (is (nil? (do-process :cook-expected-state/running :missing :custom-test-state {:state :missing
+                                                                                    :reason "Pod was explicitly deleted"
+                                                                                    :pod-deleted? true})))
+    (is (= :reason-executor-terminated @reason))))
```

Comment:
Can we put these cases up next to the existing `[running, missing]` case?

https://github.com/twosigma/Cook/pull/1540/files/67fca4eb67ae79cf7f5e234b90a9ace700fecaef..f78ac65eb64e65b431babf309653daac0c26b111#diff-e60eb787ca7fc5a6b343e52606bd47f5R80